keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -0,0 +1,32 @@
+package com.netflix.spinnaker.keel.artifact
+
+import com.netflix.spinnaker.keel.api.ArtifactType
+import com.netflix.spinnaker.keel.api.DeliveryArtifactVersion
+import com.netflix.spinnaker.keel.events.ArtifactEvent
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.net.URI
+
+@Component
+class ArtifactListener(
+  private val artifactRepository: ArtifactRepository
+) {
+  @EventListener(ArtifactEvent::class)
+  fun onArtifactEvent(event: ArtifactEvent) {
+    log.info("Received artifact event: {}", event)
+    event.artifacts.forEach {
+      val registeredArtifact = artifactRepository.get(it.name, ArtifactType.valueOf(it.type))
```

Comment:
This assumes that the delivery config with artifact info has already been saved, and so we have some concept of what artifacts we care about?

Reply:
It's checked prior to this

Reply:
Oh I see what you mean, I need to add an endpoint for that